driver/where_driver.hpp

```diff
@@ -0,0 +1,476 @@
+    int VerifyForward() override;
+    ~WhereDriver() override
+    {
+        miopenDestroyTensorDescriptor(condTensor);
+        miopenDestroyTensorDescriptor(inputTensorGrad);
+        miopenDestroyTensorDescriptor(otherTensorGrad);
+        miopenDestroyTensorDescriptor(outputTensorGrad);
+    }
+
+private:
+    InputFlags inflags;
+
+    int forw;
+    bool isInputGradRequired;
+    bool isOtherGradRequired;
+
+    // Backwards
+    miopenTensorDescriptor_t condTensor       = nullptr;
+    miopenTensorDescriptor_t outputTensorGrad = nullptr;
+    miopenTensorDescriptor_t inputTensorGrad  = nullptr;
+    miopenTensorDescriptor_t otherTensorGrad  = nullptr;
+
+    std::unique_ptr<GPUMem> cond_dev      = nullptr;
+    std::unique_ptr<GPUMem> outGrad_dev   = nullptr;
+    std::unique_ptr<GPUMem> inGrad_dev    = nullptr;
+    std::unique_ptr<GPUMem> otherGrad_dev = nullptr;
+
+    std::vector<Tgpu> cond;
+    std::vector<Tgpu> outGrad;
+    std::vector<Tgpu> inGrad;
+    std::vector<Tgpu> otherGrad;
+    std::vector<Tref> inGradhost;
+    std::vector<Tref> otherGradhost;
+};
+
+template <typename Tgpu, typename Tref>
+int WhereDriver<Tgpu, Tref>::ParseCmdLineArgs(int argc, char* argv[])
+{
+    inflags.Parse(argc, argv);
+
+    if(inflags.GetValueInt("time") == 1)
+    {
+        miopenEnableProfiling(GetHandle(), true);
+    }
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int WhereDriver<Tgpu, Tref>::GetandSetData()
+{
+    SetBNParametersFromCmdLineArgs();
+
+    std::vector<int> in_len    = GetTensorLengthsFromCmdLine("inputDims");
+    std::vector<int> other_len = GetTensorLengthsFromCmdLine("otherDims");
+    std::vector<int> cond_len  = GetTensorLengthsFromCmdLine("condDims");
+
+    isInputGradRequired = isDefined(in_len);
+    isOtherGradRequired = isDefined(other_len);
+
+    if(isInputGradRequired)
+    {
+        SetTensorNd(inputTensorGrad, in_len, data_type);
+    }
+    if(isOtherGradRequired)
+    {
+        SetTensorNd(otherTensorGrad, other_len, data_type);
+    }
+    SetTensorNd(condTensor, cond_len, data_type);
+
+    int in_sz    = in_len.size();
+    int other_sz = other_len.size();
+    int cond_sz  = cond_len.size();
+    int out_sz   = std::max({in_sz, other_sz, cond_sz});
+    std::vector<int> out_len(out_sz);
+
+    for(int i = 0; i < out_sz; i++)
+    {
+        int InVal    = (i < in_sz) ? in_len[i] : 1;
+        int OtherVal = (i < other_sz) ? other_len[i] : 1;
+        int CondVal  = (i < cond_sz) ? cond_len[i] : 1;
+        out_len[i]   = std::max({InVal, OtherVal, CondVal});
+    }
+
+    // Backwards
+    SetTensorNd(outputTensorGrad, out_len, data_type);
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int WhereDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw",
+                         'F',
+                         "1",
+                         "Run only Forward (1) or Run both Forward and Backward (0) (Default=1)",
```

Comment:
Since only backward is possible, it would be better to set the default to backward.
This is just my opinion, and you can ignore it.

Reply:
changed as per your comments